### PR TITLE
Updated docs and minor fixes

### PR DESCRIPTION
**Pull request by maheshkelkar**
- Updated README.md with Block Diagram, Concepts, Components,
  Configuration, etc
- Updated contents of some error logs
- Added code to scrub special characters from query parameters
- Added logs to dump X-Forwarded-* for debugging
- Fix the version in HealthCheck output

### CODE REVIEW

File: auth/src/main/scala/com/lookout/borderpatrol/auth/keymaster/Keymaster.scala [hunk 1]
```diff
@@ -73,12 +74,12 @@ object Keymaster {
         case Status.Forbidden => {
           statResponseDenied.incr
           Future.exception(BpIdentityProviderError(Status.Forbidden,
-            s"IdentityProvider denied user: '${req.credential.uniqueId}' with status: ${res.status}"))
+            s"IdentityProvider failed to authenticate user: '${req.credential.uniqueId}'"))
```
Comment (PeterFCaswell) [inline auth/src/main/scala/com/lookout/borderpatrol/auth/keymaster/Keymaster.scala]: Does req.credential.uniqueId contain the username? If so, you need to redact that.

File: auth/src/main/scala/com/lookout/borderpatrol/auth/keymaster/Keymaster.scala [hunk 2]
```diff
@@ -73,12 +74,12 @@ object Keymaster {
...
         }
         case _ => {
           statResponseFailed.incr
           Future.exception(BpIdentityProviderError(Status.InternalServerError,
-            s"IdentityProvider denied user: '${req.credential.uniqueId}' with status: ${res.status}"))
+            s"IdentityProvider failed to authenticate user: '${req.credential.uniqueId}', with status: ${res.status}"))
```
Comment (PeterFCaswell) [inline auth/src/main/scala/com/lookout/borderpatrol/auth/keymaster/Keymaster.scala]: Same as above.

**Comment by PeterFCaswell** [comment]
Your CI failed. :-( 
But the above are the only issues that I could see.

File: auth/src/main/scala/com/lookout/borderpatrol/auth/keymaster/Keymaster.scala (hunk 1 above)
Comment (maheshkelkar) [inline auth/src/main/scala/com/lookout/borderpatrol/auth/keymaster/Keymaster.scala]: Hi Peter, it indeed contains a username. We have a couple of more log messages that dump sensitive information such as token or username, x-forwarded-for. I will take them off as soon as we are done testing in the staging.

**Pull request merged**